VerilogCodingStyle.md

```diff
@@ -797,6 +797,92 @@ endmodule : foo
 localparam bit ValBaz = 1;  // This comment describes the item to the left.
 ```
 
+It can sometimes be useful to structure the code using header-style comments in
+order to separate different functional parts (like FSMs, the main datapath or
+registers) within a module. In that case, the preferred style is as follows:
+
+```systemverilog
+module foo;
+
+  ////////////////
+  // Controller //
+  ////////////////
+  ...
+
+  ///////////////////////
+  // Main ALU Datapath //
+  ///////////////////////
+  ...
+
+  ///////////////////////////////////////////////////////
+  // Complicated part of the design that warrants some //
+  // more description in the header comment            //
+  ///////////////////////////////////////////////////////
+  ...
+
+  ///////////////////////////////////////////////////
+  // It is ok to omit the slashes on the right side
+  ///////////////////////////////////////////////////
+  ...
+
+  case (...)
+    /////////////////////////////////////////
```

Comment:
I feel this should be just a comment. Use of subheadings should be restrictive and not over used.

Reply:
other opinions? I do not feel super strong about this, but have used it in the past...

Reply:
I felt it is enough to

```
  case (...)
    // This is ok as well, e.g. in order to
    // describe what an FSM state does in a
    // switch-case statement
    SomeState: begin
```

Reply:
sure, we can do that.

Reply:
yeah, this is aimed at a particular case: section breaks; not the general case
of commenting code.

Reply:
ok I'll remove this.